### PR TITLE
Allows contributors to give their gbp to pr authors.

### PR DESCRIPTION
**Pull request by ninjanomnom**
Just add a [💵] to your comment body to transfer the gbp. If you do not have enough gbp or you have not been added to the gbp list yet you can't give any to the author and thus the bot will edit out the command.

It's been quite a long time since I last touched php, I almost certainly fucked something up.

### CODE REVIEW

**Comment by Armhulen** [comment]
yo actually fuck yes

File: tools/WebhookProcessor/github_webhook_processor.php [hunk 1]
```diff
@@ -417,6 +420,51 @@ function create_comment($payload, $comment){
 	apisend($payload['pull_request']['comments_url'], 'POST', json_encode(array('body' => $comment)));
 }
 
+function edit_comment($payload, $new_body) {
+	apisend($payload['comment']['url'], 'PATCH', json_encode(array('body' => $new_body)))
```
Comment (ninjanomnom) [inline tools/WebhookProcessor/github_webhook_processor.php]: I'm not sure this is the correct url

**Comment by kevinz000** [comment]
please make it a special sequence :^)

**Comment by ninjanomnom** [comment]
Feel free to put suggestions for what should be the gbp transfer

**Comment by kevinz000** [comment]
[TRANSFER_GBP:<number>]?

**Comment by kevinz000** [comment]
well that worked well but you put < and >'s ?

**Comment by PKPenguin321** [comment]
Make it more specific, like [:+1:] or something

**Comment by Qustinnus** [comment]
hey you added the thing i wanted :D

**Comment by ike709** [comment]
shut up and take my GBP

**Comment by Qustinnus** [comment]
👍 have a GBP for this PR

**Comment by KorPhaeron** [comment]
People getting their friends to funnel gbp to them undermines the system especially since maintainers get gbp for merging and new accounts get bonus starting gbp.

It also is not really fair to let some people bypass the requirements to make fixes

**Comment by Armhulen** [comment]
kor i NEED this

**Comment by Qustinnus** [comment]
but people that are a positive influence to the codebase should be able to get rewarded for it.

**Comment by KorPhaeron** [comment]
You need to fix a bug!

**Comment by KorPhaeron** [comment]
@qustinnus

Fixing bugs is a positive influence and you do get rewarded for it

**Comment by Qustinnus** [comment]
people like @Armhulen probably don't have the skills to make meaningful bugfixes most likely, and you're just setting them up to literaly look for grammar problems  to get GBP. I don't think armhulen is our most negative coder but for some reason his GBP is -3

**Comment by ninjanomnom** [comment]
They do have to contribute at least once to be able to transfer gbp. 

But yeah this pr was partially to see how we all feel about this sort of feature.

**Comment by KorPhaeron** [comment]
@qustinnus

Grammar fixes are also a positive influence

**Comment by Qustinnus** [comment]
I know but forcing people that aren't good enough to make meaningful fixes to go scrounching for any grammar errors in the codebase is not really great IMO.

Why not allow us to spend our GBP to reward good PRs? Ones that we have been waiting for for a long time for example.

**Comment by KorPhaeron** [comment]
If we add a bypass to the whole system because it actually impacted one person we may as well scrap it if we think the designed impact is unfair or negative

**Comment by Qustinnus** [comment]
I always assumed this system was supposed to remove low quality PRs from the codebase, not discourage newbies.

**Comment by Armhulen** [comment]
i joined coding when the GBP reserve was at it's lowest

**Comment by KorPhaeron** [comment]
You only spend gbp by having your pr merged so it doesnt actually do anything to prevent people opening bad ones

**Comment by KorPhaeron** [comment]
And again, if you think the primary impact of this system is negative and only serves to discourage newbies we should be talking about disabling it rather than having a bypass for people with enough friends

**Comment by KorPhaeron** [comment]
@cyberboss

Thoughts on this please

**Comment by ninjanomnom** [comment]
In light of what you've been talking about it feels like a better solution is not to have negative points at all in tags and the positives should be nerfed.

Instead negative gbp could be assigned by maintainers in issues by blaming bugs on merged prs or in the body of the pr if it has the tested/reproduced label.

**Comment by duncathan** [comment]
please make balance still worth negatives

**Comment by ninjanomnom** [comment]
Dunno if I feel like working on it atm but maybe later. It's the same complexity to implement as this

**Comment by Cyberboss** [comment]
I don't like this. Certain people make enough GBP to support every contributor 10x over. But, even so, alone these people do not stem the ever growing tide of issues. The fact is maintainers alone can't fix the game, we rely on contributors to help with that. This change goes against that core idea that created the PR balance.

**Comment by Cyberboss** [comment]
I'd be less opposed if this was limited to maintainers only. However, then we run the inevitable risk of playing favoritism in an already controversial system.

**Comment by Cyberboss** [comment]
You can ask Zbot in #coderbus to open any random dm file, and you can find literally the smallest thing to improve for 1 GBP. I've done this countless times myself. If you need an example just ping me and I can point something out.

**Comment by PKPenguin321** [comment]
I like this if only because I can now keep a PR open by just donating a GBP instead of manually remaking the PR

**Comment by KorPhaeron** [comment]
@Cyberboss Would it be at all feasible to reward people like .5 gbp for testing/tagging old and outdated issues and getting them closed? Bribe people to clean the issue tracker?

**Comment by Cyberboss** [comment]
I'd give them a whole gbp for each highlighted issue tbh. It's a pretty good job

**Comment by KorPhaeron** [comment]
Sure

**Comment by Qustinnus** [comment]
@Armhulen son you'll be rich if you do that

**Pull request closed without merging**